src/internal/operators/reduce.ts

```diff
@@ -6,9 +6,9 @@ import { OperatorFunction, MonoTypeOperatorFunction } from '../types';
 import { pipe } from '../util/pipe';
 
 /* tslint:disable:max-line-length */
+export function reduce<T, R>(accumulator: (acc: R, value: T, index: number) => R, seed: R): OperatorFunction<T, R>;
```

Comment:
So there's this messed up scenario I think we need to test for:

`of(1).pipe(reduce((acc, value, i) => i === 0 ? [] : [..value], { seed: 'LOL' }))` The problem is that `acc` gets typed wrong.

I'm sure that `scan` has the same issue.

Reply:
So we assume that `seed` and `ReturnType<typeof accumulator>` are the same type, if they differ we get an error because accumulator returns an invalid value.  Thoughts?